### DISCUSSION TITLE
How to run with docker?

### DISCUSSION THREAD

**Discussion by ranpin** [IDEA-Research/Grounded-Segment-Anything#199] [Q&A]
I'm new to docker and I don't quite understand the **_Open one terminal:make run That's it._** written in readme.md. Can someone please explain it in detail? Also, I can't find the image for this project in Docker Desktop, what keyword should I enter?

 Or is there something I need to learn to help me get the project running? Thanks in advance for your advice!
![image](https://user-images.githubusercontent.com/51260021/233971493-02333464-b0ae-4adc-ac5f-02b38325efc7.png)

**Comment by evanshlom** [comment]
If you were able to create a virtual environment like with conda and you got the program to run that way, you can export the dependencies from the venv to a requirements.txt then you can refer to that reqs file in a Docker file. The rest would be the usual workflow of using a docker image to run a container...to figure out the general docker workflow you can consult the docker docs or an article or YouTube video. Basically you run a terminal command to build the image from the docker file and then you run the program using the container.
